components/fetch-build-scan-data-cmdline-tool/build.gradle.kts

```diff
@@ -29,11 +35,32 @@ java {
     }
 }
 
+swaggerSources.register("gradleEnterprise") {
+    setInputFile(layout.projectDirectory.file("src/main/openapi/openapi.yaml").asFile)
+    code.apply {
+        language = "java"
+        configFile = layout.projectDirectory.file("src/main/openapi/openapi-generator-config.json").asFile
+    }
+}
+
+sourceSets {
+    main {
+        java {
+            srcDir(layout.buildDirectory.dir("swagger-code-gradleEnterprise/src/main/java"))
+        }
+    }
+}
+
 tasks.compileJava {
+    dependsOn("generateSwaggerCode")  // TODO figure out how to eliminate the explicit dependency
```

Comment:
Is using the output of the `generateSwaggerCode` task as the input to the srcDir not possible (and avoid this explicit task dependency)?

See [here](https://github.com/etiennestuder/gradle-jooq-plugin/blob/master/src/main/groovy/nu/studer/gradle/jooq/JooqPlugin.java#L54) for what I do in the gradle-jooq-plugin. This is also an item to be handled with low prio post-release.

Reply:
✅ Done. Something along these lines worked rather nicely.